api/routes/tag_routes.py

```diff
@@ -3,28 +3,40 @@
 from db.queries.tags.queries import insert_tags
 from db.queries.tags.queries import select_tags_for_fund_round
 from db.queries.tags.queries import select_tags_types
+from db.queries.tags.queries import update_tags
 from db.schemas.schemas import JoinedTagSchema
 from db.schemas.schemas import TagSchema
 from db.schemas.schemas import TagTypeSchema
 from flask import abort
 from flask import current_app
 from flask import request
+from flask import Response
 
 
-def get_tags_for_fund_round(fund_id, round_id):
+def get_tags_for_fund_round(
+    fund_id,
+    round_id,
+    tag_purpose: str = "ALL",
+    tag_status: bool = True,
+    search_term: str = "",
+    search_in: str = None,
+):
+    tags = select_tags_for_fund_round(
+        fund_id, round_id, tag_purpose, tag_status, search_term, search_in
+    )
 
-    tags = select_tags_for_fund_round(fund_id, round_id)
+    if tags:
+        serialiser = JoinedTagSchema()
+        serialised_tags = [serialiser.dump(r) for r in tags]
+        return serialised_tags
 
-    if not tags:
-        current_app.logger.warning(
-            f"No tags found for fund__round: {fund_id}__{round_id}."
-        )
-    serialiser = JoinedTagSchema()
-    serialised_tags = [serialiser.dump(r) for r in tags]
-    return serialised_tags
+    return Response(
+        response=f"No tags found for fund__round: {fund_id}__{round_id}.",
+        status=204,
+    )
```

Comment:
I get this has been handled on the front-end, but shouldn't we just return an empty list?

Reply:
true ill return an empty list here instead